.infrastructure/identity/main.tf

```diff
@@ -122,3 +122,48 @@ resource "aws_iam_role_policy_attachment" "githubiac" {
   role       = aws_iam_role.githubiac.name
   policy_arn = data.aws_iam_policy.admin_access.arn
 }
+
+###############################################################################
+#                      Define IAM Role to use on deploy                       #
+###############################################################################
+data "aws_iam_policy" "deploy_access" {
+  name = "AmazonS3FullAccess"
+}
+
+resource "aws_iam_role" "githubaction_deploy" {
+  name        = "GitHubActionDeployRole"
+  description = "Role to assume to deploy the website"
+
+  assume_role_policy = jsonencode({
+    "Version": "2012-10-17",
+    "Statement": [
+      {
+        "Effect": "Allow",
+        "Action": [
+          "s3:ListBucket",
+          "s3:PutObject",
+          "s3:GetObject",
+          "s3:DeleteObject"
+        ],
+        "Resource": [
+          "arn:aws:s3:::*",
+          "arn:aws:s3:::*/*"
```

Comment:
Following the [principle of least privilege](https://en.wikipedia.org/wiki/Principle_of_least_privilege) we can [restrict the resource providing the bucket arn](https://github.com/pagopa/new-corporate-site-infrastructure/blob/main/src/main/05-iam.tf#L160-L205).